### PR TITLE
[ADD] init l10n_ir module

### PR DESCRIPTION
**Pull request by saeed-raeisi**
add l10n_ir module
accounting and states data

### CODE REVIEW

**Comment by saeed-raeisi** [comment]
https://github.com/odoo/odoo/pull/97784

**Comment by saeed-raeisi** [comment]
`l10n_ir/data/l10n_ir_chart_data.xml:3: [W7902(duplicate-xml-record-id), ]  Duplicate xml record id "data/l10n_ir_chart_template_noupdate_0" in data/l10n_ir_chart_post_data.xml:3`


add new "account.chart.template"   l10n_ir_chart_data.xml:
<?xml version="1.0" encoding="utf-8" ?>
```
<odoo>
    <record id="l10n_ir_chart_template" model="account.chart.template">
        <field name="name">Iran Chart Of Account</field>
          ...
    </record>
</odoo>
```



use in 
account.account.template.csv
```
id,name,code,user_type_id/id,chart_template_id/id,reconcile
ir_account_111001,rial fund,111001,account.data_account_type_liquidity,l10n_ir.l10n_ir_chart_template,1
```

after:
update id="**l10n_ir_chart_template**"  l10n_ir_chart_post_data.xml
file :
```
<odoo>
    <record id="l10n_ir_chart_template" model="account.chart.template">
        <field name="property_account_receivable_id" ref="ir_account_111411" />
       ....
    </record>
</odoo>
```

`l10n_ir/data/l10n_ir_chart_data.xml:3: [W7902(duplicate-xml-record-id), ]  Duplicate xml record id "data/l10n_ir_chart_template_noupdate_0" in data/l10n_ir_chart_post_data.xml:3`

in odoo module : l10n_jp is same this 
https://github.com/odoo/odoo/blob/16.0/addons/l10n_jp/data/l10n_jp_chart_data.xml
https://github.com/odoo/odoo/blob/16.0/addons/l10n_jp/data/account.account.template.csv
https://github.com/odoo/odoo/blob/16.0/addons/l10n_jp/data/account_chart_template_data.xml

**Comment by saeed-raeisi** [comment]
W7902(duplicate-xml-record-id)
@pedrobaeza 
This is also the case in these repositories





https://github.com/OCA/l10n-usa/blob/84fc37cb8cf77bc5800191834a0066984ee1017b/l10n_us_gaap/data/account_chart_template_data.xml
https://github.com/OCA/l10n-usa/blob/84fc37cb8cf77bc5800191834a0066984ee1017b/l10n_us_gaap/data/account_us_gaap_data.xml

l10n_us_gaap_chart_template



https://github.com/OCA/l10n-france/blob/6e88a15a3c91f2ac1a345511e3ed28e721d8e65e/l10n_fr_oca/data/l10n_fr_chart_data.xml
https://github.com/OCA/l10n-france/blob/6e88a15a3c91f2ac1a345511e3ed28e721d8e65e/l10n_fr_oca/data/account_chart_template_data.xml

l10n_fr_pcg_chart_template

**Comment by saeed-raeisi** [comment]
.pylintrc - .pylintrc-mandatory :
removed
`duplicate-xml-record-id`

.pre-commit-config.yaml : 
removed
`id: oca-checks-odoo-module`



--- update:

`<!-- pylint:disable=duplicate-xml-record-id -->`

**Comment by saeed-raeisi** [comment]
https://github.com/OCA/l10n-iran/blob/dev-15.0/l10n_ir/data/account.account.template.csv
https://github.com/OCA/l10n-iran/blob/dev-15.0/l10n_ir/data/account.group.template.csv

this files need review translation From Fa to En
and this file
https://github.com/OCA/l10n-iran/blob/dev-15.0/l10n_ir/i18n/fa.po

**Comment by pedrobaeza** [comment]
Hi @saeed-raeisi

The XML-ID duplicated is a valid check for most of the occasions, as it prevents to declare 2 times the same record. The only exception is this needed trick where you need first to create the CoA, and then, assign an account.

It's better to just disable locally before the record declaration than to disable it totally in the configuration. Something like:
```
<!-- pylint: disable=duplicate-xml-record-id -->
```

**Comment by saeed-raeisi** [comment]
> Hi @saeed-raeisi
> 
> The XML-ID duplicated is a valid check for most of the occasions, as it prevents to declare 2 times the same record. The only exception is this needed trick where you need first to create the CoA, and then, assign an account.
> 
> It's better to just disable locally before the record declaration than to disable it totally in the configuration. Something like:
> ```
> 
> ```

Hi
 This issue was fixed in the last commit

File: .flake8 [hunk 1]
```diff
@@ -7,6 +7,6 @@ select = C,E,F,W,B,B9
 # E203: whitespace before ':' (black behaviour)
 # E501: flake8 line length (covered by bugbear B950)
 # W503: line break before binary operator (black behaviour)
-ignore = E203,E501,W503
+ignore = E203,E501,W503,W7902
```
Comment (pedrobaeza) [inline .flake8]: I still see this as general ignore instead of local one.

**Comment by pedrobaeza** [comment]
OK, now you can squash all the commits into one, and I think it's OK to merge it.

**Comment by pedrobaeza** [comment]
You haven't squashed the commits, Saeed. Please take it into account for other times.

**Comment by saeed-raeisi** [comment]
> You haven't squashed the commits, Saeed. Please take it into account for other times.

Sorry,
I thought ocabot could do that
Is it possible to choose merge method  for ocabot?

**Comment by pedrobaeza** [comment]
No, it's not possible.

**Pull request merged**